### ISSUE TITLE
OneCycleLR

### ISSUE THREAD

**Issue by Some-random** [espnet/espnet#3019]
Hi, I've read about OneCycleLR in this [paper](https://arxiv.org/abs/2010.13956) about Conformer in ESPNET and it looks like ESPNET has already implemented [it](https://github.com/espnet/espnet/blob/dab2092bc9c8e184c48cc6e603037333bd97dcd1/espnet2/schedulers/abs_scheduler.py#L81). I'm wondering how big of an improvement has one observed using OneCycleLR (in terms of training speed and final accuracy, for any task). Also if we want to use it, do we need to perform a learning rate range test first?

**Comment by sw005320** [comment]
@pengchengguo, could you report it?
I think it is a good idea to record such trials in our issues.

**Comment by pengchengguo** [comment]
Hi @Some-random, thanks for your interest in our paper. Actually, we didn't tune many parameters and the OneCycleLR showed almost the same results as Warmup scheduler. Here are some results on Aishell-1 corpus.

| Model | Dev | Test |
| -- | -- | -- |
| Conformer (Warmup) | 4.7 | 5.2 |
| Conformer (OneCycleLR) | 4.6 | 5.1 |

It should be noted that both models are trained on ESPnet1 with `torch 1.6` and `builtin CTC loss`, so the results of Warmup are slightly worse than our posted results (https://github.com/espnet/espnet/blob/master/egs/aishell/asr1/RESULTS.md).

As for the training speed, the Warmup model costs 5 days on 1 GPU, while the OneCycleLR costs 7 days on 1 GPU. Since we trained them on a cluster, the training speed was also influenced by other running tasks. Here are some parameters of our OneCycleLR model.

![image](https://user-images.githubusercontent.com/23135280/109384905-f6bb7600-792a-11eb-9bb2-73120bf3c3d1.png)
